### ISSUE TITLE
bug: can't infer superclass constant

### ISSUE THREAD

**Issue by SolaWing** [castwide/solargraph#281]
```ruby
class A
  class AA
  end
end

class B < A
  def f
    a = AA.new # AA can't infer
  end
end

pp B.new().f
```

**Comment by castwide** [comment]
Confirmed. This appears to be a simple issue in constant detection. It's on the roadmap.

**Comment by castwide** [comment]
Fixed in the v0.39 branch (https://github.com/castwide/solargraph/commit/b611c5a8f8cc0243ab9d1b4540e848b3dbf47b9a).
